src/Orchard.Environment.Shell.Data/ShellDescriptorManager.cs

```diff
@@ -31,14 +32,20 @@ public class ShellDescriptorManager : IShellDescriptorManager
             _eventBus = eventBus;
             _session = session;
             _logger = logger;
+
+            _asyncLazyShellDescriptor = new Lazy<Task<ShellDescriptor>>(() => Task.Run(() =>
```

Comment:
Oh yes i understand, so it's more than just using `AllFeaturesShellDescriptorManager` with no db access, because there are still db access during shell context creation i've added.

But anyway currently to run modules startups we need to create a shell context, so maybe we have to use custom shell factories. Need to think about this, i've no immediate solution.